### PR TITLE
fix(tag-status)!: replace tag status props to expand functionallity

### PR DESCRIPTION
**Pull request by GoncaloCamaz**
No visual changes were done to the component.
However, to use this component, the user is required to specify the icon to be displayed; the color of the icon defined by the state prop. The label is optional

### CODE REVIEW

**Review by djcouto — APPROVED** [review]
I like the approach 👍 

I'll also think would be valuable if we create an `AssetTagStatus`, `ClusterTagStatus`, etc, that reuses this component and has an icon and label configs mapped by a state. This way we wouldn't need to replicate these configs every time we need to use this component for those particular use cases. 

But since this is just a personal opinion I'll approve this.

**Comment by ricardogoncalves89** [comment]
> I'll also think would be valuable if we create an `AssetTagStatus`, `ClusterTagStatus`, etc, that reuses this component and has an icon and label configs mapped by a state. This way we wouldn't need to replicate these configs every time we need to use this component for those particular use cases.

@djcouto I think it's more application scope and not design system scope

**Comment by djcouto** [comment]
> > I'll also think would be valuable if we create an `AssetTagStatus`, `ClusterTagStatus`, etc, that reuses this component and has an icon and label configs mapped by a state. This way we wouldn't need to replicate these configs every time we need to use this component for those particular use cases.
> 
> @djcouto I think it's more application scope and not design system scope

You're absolutely right, It doesn't belong in our design system 👍

**Review by ricardogoncalves89 — APPROVED** [review]
Please add the `BREAKING CHANGE:` inside the commit message with what you are breaking

File: packages/ui-components/src/components/tag-status/tag-status.tsx [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
 	tag: 'kv-tag-status',
 	styleUrl: 'tag-status.scss',
 	shadow: true
 })
 export class KvTagStatus implements ITagStatus {
-	@Prop({ reflect: true }) type: ETagStatusType = ETagStatusType.Unknown;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) state: ETagState;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) icon: EIconName | EOtherIconName;
```
Comment (pedrorebelocosta) [inline packages/ui-components/src/components/tag-status/tag-status.tsx:15]: should we make this property non-null using the `!` operator?

**Review by pedrorebelocosta — APPROVED** [review]

**Pull request merged**